src/rprblender/engine/viewport_engine_2.py

```diff
@@ -1,58 +1,36 @@
-#**********************************************************************
-# Copyright 2020 Advanced Micro Devices, Inc
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#     http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-#********************************************************************
+import numpy as np
+import math
+import concurrent.futures
 import time
 import threading
-
 import pyrpr
-
 from .viewport_engine import ViewportEngine, ViewportSettings, FinishRenderException
 from .context import RPRContext2
-
 from rprblender.utils import logging
-log = logging.Log(tag='viewport_engine_2')
 
+log = logging.Log(tag='viewport_engine_2')
 
 class ViewportEngine2(ViewportEngine):
     _RPRContext = RPRContext2
 
     def __init__(self, rpr_engine):
         super().__init__(rpr_engine)
-
         self.is_last_iteration = False
         self.rendered_image = None
-
         self.resolve_event = threading.Event()
         self.resolve_thread = None
         self.resolve_lock = threading.Lock()
 
     def stop_render(self):
-        self.is_finished = True
-        self.restart_render_event.set()
-        self.resolve_event.set()
-        self.sync_render_thread.join()
-        self.resolve_thread.join()
+        super().stop_render()  # Call parent stop_render method
```

Comment:
Need additional testing... It feels like this was some kind of bug workaround in a past (but I am not sure)